src/main/java/cd/connect/openapi/DartV3ApiGenerator.java

```diff
@@ -217,8 +224,46 @@ private void correctInternals(CodegenModel model, CodegenProperty cp, boolean cl
       cp.isAnyType = true;
       cp.isPrimitiveType = true;
     }
-
-    if (cp.allowableValues != null && cp.allowableValues.get("enumVars") != null) {
+    // detect wether the schema follows the pattern:
+    // "{ "nullable" : true, "allOf" : [ { "$ref" : <SomeEnum> } ]}",
+    // if yes, treat it like an enum.
+    // this is a fix for https://github.com/dart-ogurets/dart-openapi-maven/issues/48
+    boolean isComposedEnum = false;
+    if (cp.isModel) {
+      try {
+        ObjectNode schemaNode = (ObjectNode) ObjectMapperFactory.createJson().readTree(cp.jsonSchema);
+        Schema<?> parsedSchema = new OpenAPIDeserializer().getSchema(schemaNode, "", null);
+        if (parsedSchema instanceof ComposedSchema) {
+          ComposedSchema composedSchema = (ComposedSchema) parsedSchema;
+          @SuppressWarnings("rawtypes") // cannot be fixed unless upstream changes its signatures to Schema<?>
+          List<Schema> allOfArray = composedSchema.getAllOf();
+          @SuppressWarnings("rawtypes") // cannot be fixed unless upstream changes its signatures to Schema<?>
+          List<Schema> oneOfArray = composedSchema.getOneOf();
+          @SuppressWarnings("rawtypes") // cannot be fixed unless upstream changes its signatures to Schema<?>
+          List<Schema> anyOfArray = composedSchema.getAnyOf();
+          if (allOfArray.size() == 1 && anyOfArray == null || anyOfArray.isEmpty() && oneOfArray == null
+              || oneOfArray.isEmpty()) {
+            String singleRef = allOfArray.get(0).get$ref();
+            if (singleRef != null) {
+              Matcher extractLastPathComponent = Pattern.compile("#\\/components\\/schemas\\/([a-zA-Z0-9\\.\\-_]+)")
+                  .matcher(singleRef);
+              if (extractLastPathComponent.matches()) {
+                List<?> referencedEnum = openAPI.getComponents().getSchemas().get(extractLastPathComponent.group(1))
+                    .getEnum();
+                if (referencedEnum != null && !referencedEnum.isEmpty()) {
+                  isComposedEnum = true;
+                  cp.isModel = false;
+                }
+              }
+            }
+          }
+        }
+      } catch (Exception e) {
+        e.printStackTrace();
```

Comment:
this needs to be getLog().error("....", e)

Reply:
ah... I had that there for debugging purposes and didn't really intend to log anything there in the end, but just assume that the schema isn't of the form described and silently continue as normal. But now that I'm catching most errors that could come up anyway with the if statements it may make sense to log something there

Reply:
alright, how about this?